devito/ops/operator.py

```diff
@@ -108,13 +109,13 @@ def _specialize_iet(self, iet, **kwargs):
 
     @property
     def hcode(self):
-        return ''.join(str(kernel) for kernel in self._ops_kernels)
+        return ''.join(str(kernel.root) for kernel in self._func_table.values())
 
     def _finalize(self, iet, parameters):
         # Applies _finalize for each generated OPS kernel, which will generate the .h file
         self._ops_kernels = [super(OperatorOPS, self)._finalize(
```

Comment:
this PR looks broken, or your testing is broken in a bad way

here, there's still an occurrence of `_ops_kernels`  and nothing is failing

Reply:
Thats because I was using the `func_table` as a temporary holder, and in the end I was still using the `_ops_kernels` list as you have pointed in here.

I have pushed a fix now. It seems to be working, and I generate the code locally and the .h file seems to be ok.